src/storm/solver/helper/OptimisticValueIterationHelper.cpp

```diff
@@ -52,29 +52,34 @@ namespace storm {
+                    return singleIterationWithDiffInternal<false, storm::solver::OptimizationDirection::Minimize>(x, b, computeRelativeDiff, schedulerFixedForRowgroup, scheduler);
                 }
                 
                 template <typename ValueType>
-                ValueType IterationHelper<ValueType>::singleIterationWithDiff(storm::solver::OptimizationDirection const& dir, std::vector<ValueType>& x, std::vector<ValueType> const& b, bool computeRelativeDiff) {
+                ValueType IterationHelper<ValueType>::singleIterationWithDiff(storm::solver::OptimizationDirection const& dir, std::vector<ValueType>& x, std::vector<ValueType> const& b, bool computeRelativeDiff, boost::optional<storage::BitVector>  const& schedulerFixedForRowgroup, boost::optional<std::vector<uint_fast64_t>>  const& scheduler) {
                     if (minimize(dir)) {
-                        return singleIterationWithDiffInternal<true, storm::solver::OptimizationDirection::Minimize>(x, b, computeRelativeDiff);
+                        return singleIterationWithDiffInternal<true, storm::solver::OptimizationDirection::Minimize>(x, b, computeRelativeDiff, schedulerFixedForRowgroup, scheduler);
                     } else {
-                        return singleIterationWithDiffInternal<true, storm::solver::OptimizationDirection::Maximize>(x, b, computeRelativeDiff);
+                        return singleIterationWithDiffInternal<true, storm::solver::OptimizationDirection::Maximize>(x, b, computeRelativeDiff, schedulerFixedForRowgroup, scheduler);
                     }
                 }
                 
                 template <typename ValueType>
                 template<bool HasRowGroups, storm::solver::OptimizationDirection Dir>
-                ValueType IterationHelper<ValueType>::singleIterationWithDiffInternal(std::vector<ValueType>& x, std::vector<ValueType> const& b, bool computeRelativeDiff) {
+                ValueType IterationHelper<ValueType>::singleIterationWithDiffInternal(std::vector<ValueType>& x, std::vector<ValueType> const& b, bool computeRelativeDiff, boost::optional<storage::BitVector>  const& schedulerFixedForRowgroup, boost::optional<std::vector<uint_fast64_t>>  const& scheduler) {
                     STORM_LOG_ASSERT(x.size() > 0, "Empty equation system not expected.");
                     ValueType diff = storm::utility::zero<ValueType>();
                     
                     IndexType i = x.size();
                     while (i > 0) {
                         --i;
-                        ValueType newXi = HasRowGroups ? multiplyRowGroup<Dir>(i, b, x) : multiplyRow(i, b[i], x);
+                        ValueType newXi;
+                        if (schedulerFixedForRowgroup && schedulerFixedForRowgroup.get()[i]) {
+                            newXi = multiplyRow(i, b[i + scheduler.get().at(i)], x);
```

Comment:
This currently multiplies the `i`th row, right? shouldn't it be something like
```suggestion
                            auto row = i + scheduler.get().at(i);
                            newXi = multiplyRow(row, b[row], x);
```
same applies to similar occurrences below.